app/models/declaration/submissions/CancellationStatus.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.declaration.submissions
+
+import play.api.libs.json._
+
+sealed trait CancellationStatus
+
+object CancellationStatus {
+
+  def unapply(status: CancellationStatus): Option[(String, JsValue)] = {
+    val (prod: Product, sub) = status match {
+      case CancellationRequestExists =>
+        (CancellationRequestExists, Json.toJson(CancellationRequestExists.toString))
```

Comment:
Maybe case x => (x, JsString(x.toString)) so we don’t need a case for each option

Reply:
This is the same as previous one: case objects are a bit tricky and you can't avoid multiple cases. At least I am not aware of any way to do it differently.